lxd/firewall/drivers/drivers_consts.go

```diff
@@ -51,5 +51,7 @@ type AddressForward struct {
 	TargetAddress net.IP
 	Protocol      string
 	ListenPorts   []uint64
```

Comment:
Can you remove the `ListenPorts` and `TargetPorts` fields so we don't pass around duplicate info?

Reply:
I did do this initially but it caused an issue in https://github.com/lxc/lxd/blob/master/lxd/network/driver_bridge.go#L2296 and I was a bit concerned about making too many changes. 

Shall I go ahead with it? I'll need to convert the listen and target ports in https://github.com/lxc/lxd/blob/master/lxd/network/driver_common.go#L38 to the ranges we need.

Reply:
You can do it as a separate set of commits on this PR once you've got the actual fix working and tested.

Reply:
Ultimately we are trying to stop passing port lists around and start passing port ranges around so the recipients of those port ranges can use them more efficiently where possible, and where not possible just expand them back to port lists. You may even find that adding an ExpandTargetPorts() function on the AddressForward struct might be useful if you find you're calling it repeatedly in multiple places in the code.

Reply:
Ok sounds good.

Reply:
I have removed `PortRanges` in favour of using the original `Ports` field. See https://github.com/lxc/lxd/pull/9614#discussion_r762862079.